python_modules/libraries/dagster-components/dagster_components/lib/dbt_project/component.py

```diff
@@ -25,49 +27,54 @@
 from dagster_components.utils import TranslatorResolvingInfo, get_wrapped_translator_class
 
 
-class DbtProjectParams(ComponentSchema):
+class DbtProjectSchema(ResolvableSchema["DbtProjectComponent"]):
     dbt: DbtCliResource
     op: Optional[OpSpecSchema] = None
     asset_attributes: Annotated[
-        Optional[AssetAttributesSchema], ResolvableFieldInfo(required_scope={"node"})
+        Optional[AssetAttributesSchema],
+        ResolvableFieldInfo(required_scope={"node"}),
     ] = None
     transforms: Optional[Sequence[AssetSpecTransformSchema]] = None
 
 
-@resolver(fromtype=DbtProjectParams, exclude_fields={"asset_attributes"})
-class DbtProjectResolver(Resolver[DbtProjectParams]):
-    def resolve_translator(self, context: ResolutionContext) -> DagsterDbtTranslator:
+@registered_component_type(name="dbt_project")
+@dataclass
+class DbtProjectComponent(Component):
+    """Expose a DBT project to Dagster as a set of assets."""
+
+    dbt: DbtCliResource
+    op: Optional[OpSpecSchema]
+    translator: DagsterDbtTranslator
+    transforms: Optional[Sequence[Callable[[Definitions], Definitions]]] = None
+
+    @field_resolver("dbt")
+    @staticmethod
+    def resolve_dbt(schema: DbtProjectSchema, context: ResolutionContext) -> DbtCliResource:
+        return DbtCliResource(**context.resolve_value(schema.dbt.model_dump()))
+
+    @field_resolver("translator")
+    @staticmethod
+    def resolve_asset_attributes(
```

Comment:
seems like our convention should be `resolve_translator` since we are computing the translator here